### DISCUSSION TITLE
Onchain tests used to gain protocol trust

### DISCUSSION THREAD

**Discussion by rawsilence** [sablier-labs/evm-monorepo#689] [Show and tell]
I wrote 2 onchain tests that can be deployed and staked on [Ante Finance](https://docs.ante.finance/ante/getting-started/how-ante-works).
1. Tests that Sablier TVL of USDC + WETH + DAI doesn't drop more than 90% over time since the time of deployment
https://github.com/antefinance/ante-community-tests/pull/228

2. Tests that Sablier admin will never set the protocol fee to more than `MAX_FEE` (10%). Doing that will cause all new streams to fail being created.
https://github.com/antefinance/ante-community-tests/pull/229

This is a way Sablier can bet (stake) on its own security and gain more trust from users.

**Comment by PaulRBerg** [comment]
Thanks @danielcrt. Ante Finance is an interesting protocol!

The 2nd test is awesome and definitely something that we would be keen to vouch for. However, I am on the fence about the 1st test related to the TVL. Could you share more details about the rationale?

Why would we make a financial bet against our users' volume in one particular release of the contracts? If the bear market continues for a long time, we will lose the stake 😅

**Reply by rawsilence** [reply]
I think it's a powerful statement to say that the protocol won't go to (near) 0 during a bear market and this can build trust for potential new clients.
If a project's TVL is heading to 0 changes for it to go bankrupt are higher given the lack of interest.

I agree that this specific test might not be appropriate for Sablier at this point in time given the low volumes for the tested tokens and the high variety of tokens your clients are using (e.g. vesting their own tokens)

**Reply by PaulRBerg** [reply]
I see, thanks for the additional color.

We will skip on this TVL test for the time being for the reasons you have given. Also because we expect to have a large variety of contracts (we're gonna make a new deployment later this year) and so the TVL will spread out pretty thin.

But the fee guarantee is cool. We will have a look!
